src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -140,107 +139,77 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
                 return;
             }
 
+            try
+            {
+                await ProcessMessage(eventArgs).ConfigureAwait(false);
+            }
+            finally
+            {
+                semaphore.Release();
+            }
+        }
+
+        private async Task ProcessMessage(BasicDeliverEventArgs message)
+        {
             Dictionary<string, string> headers = null;
             string messageId = null;
-            var pushMessage = false;
-
             try
             {
                 try
                 {
                     messageId = messageConverter.RetrieveMessageId(message);
                     headers = messageConverter.RetrieveHeaders(message);
-
-                    pushMessage = true;
                 }
                 catch (Exception ex)
                 {
                     await poisonMessageForwarder.ForwardPoisonMessageToErrorQueue(message, ex, settings.ErrorQueue).ConfigureAwait(false);
+                    await TryAcknowledgeMessage(message.DeliveryTag, messageId).ConfigureAwait(false);
```

Comment:
Formatting: I'd prefer to have an empty line between the `TryAcknowledgeMessage` call and the `return`.